### DISCUSSION TITLE
Where is the definition for the `strapi` object?

### DISCUSSION THREAD

**Discussion by joeyfezster** [strapi/strapi#8038] [Legacy Topics]
Hi all,

Not wanting to come of as a total noob, but I keep seeing an object `stapi` referenced in tons of places as if it was there, but not visible in any import.

An example can be seen in [this file](https://github.com/strapi/strapi/blob/35aafde2288e6858a9b1757cf2555a8561421140/packages/strapi-admin/services/user.js#L42), line 42, where it calls ` return strapi.query('user', 'admin').create(user);`. There's no `strapi` imported or anything, and vs code does not recognize any definition for the object. 

Moreover, searching for it is obviously a huge pain, as the word is highly repetitive 😄 .

**Where is it defined, and what is the mechanism that allows for something like this to even be done?**

Thanks

**Comment by derrickmehaffy** [comment]
I can certainly see why this would be confusing :laughing: due to how Strapi is loaded, no import is required as it's a global API (though it does make it a little difficult since there is no intellisense) The documentation for it is described here: https://strapi.io/documentation/v3.x/global-strapi/api-reference.html

However it isn't entirely documented yet :see_no_evil: and we know that can be a huge blocker for those new to the framework, we do plan to address this in the future (both documentation and intellisense) but no ETA for the moment on that.

**Reply by derrickmehaffy** [reply]
You can also use the command `yarn strapi console` to get an interactive console and explore this internal API

**Reply by joeyfezster** [reply]
Thanks @derrickmehaffy!

So I guess [this](https://github.com/strapi/strapi/blob/master/packages/strapi/lib/Strapi.js#L441) would be where the global `starpi` object is defined then. For posterity :D

**Reply by derrickmehaffy** [reply]
Technically yes :laughing: it's a bit of an odd design (some technical debt involved) but we are looking to make it a bit easier in the future.
